server/src/main/java/edu/jhu/server/Game.java

```diff
@@ -52,17 +66,93 @@ public void notifyPlayers(JSONObject event) {
       player.sendEvent(event);
     }
   }
-
+  
+  private JSONObject makeChatMessage(String body) {
+	  JSONObject chat = new JSONObject();
+	  chat.put("eventType", "CHAT_NOTIFICATION");
+	  chat.put("author", "Game");
+	  chat.put("body", body);
+	  return chat;
+  }
+  
+  private JSONObject makeInvalidRequestMessage(String player, String reason){
+	  JSONObject invalidRequest = new JSONObject();
+	  invalidRequest.put("eventType", "INVALID_REQUEST_NOTIFICATION");
+	  invalidRequest.put("author", "Game");
+	  invalidRequest.put("reason", reason);
+	  invalidRequest.put("player", player);
+	  return invalidRequest;
+  }
+  
+  private JSONObject makeMoveNotification(Suspect suspect, ILocation location){
+	  JSONObject move = new JSONObject();
+	  move.put("eventType", "MOVE_NOTIFICATION");
+	  return move;
+  }
+  
+  
+  
+  private void provideEvidence(CaseFile casefile, Player suggester) {
+	  Player playerWithEvidence = null;
+	  for (Player player : players) {
+		  if(player.getTag() != suggester.getTag() && 
+			(player.hasCard(casefile.getRoom()) || 
+			player.hasCard(casefile.getSuspect()) ||
+			player.hasCard(casefile.getWeapon()))) {
+			  playerWithEvidence = player;
+			  break;
+		  }  
+	  }
+	  if (playerWithEvidence == null) {
+		  JSONObject chat = makeChatMessage("Nobody could provide evidence against this suggestion!");
+		  handleEvent(chat);
```

Comment:
in general, i think handleEvent() should be only used for incoming events from the frontend, and notifyPlayers() should be used to send an event to the frontend